validator_test.go

```diff
@@ -0,0 +1,273 @@
+/**
+ *  Copyright (c) 2022  Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package wrp
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func testTypeValidatorValidate(t *testing.T) {
+	type Test struct {
+		m                 map[MessageType]Validator
+		defaultValidators Validators
+		msg               Message
+	}
+
+	tests := []struct {
+		description string
+		value       Test
+		expectedErr error
+	}{
+		// Success case
+		{
+			description: "Found success",
+			value: Test{
+				m: map[MessageType]Validator{
+					SimpleEventMessageType: Validators{AlwaysValid},
```

Comment:
It's worth noting that `Validators` isn't needed here.  I would remove `Validators` from this test entirely, so we cannot ever rely on `Validators` functionality in our testing of `TypeValidator`.